File: MFDLabs.Grid.Bot/Commands/OpenJob.cs
using System.Linq;
using System.Threading.Tasks;
using Discord.WebSocket;
using MFDLabs.Grid.Bot.Extensions;
using MFDLabs.Grid.Bot.Interfaces;
using MFDLabs.Grid.Bot.Utility;

namespace MFDLabs.Grid.Bot.Commands
{
    internal class OpenJob : IStateSpecificCommandHandler
    {
        public string CommandName => "Open Grid Server Job";
        public string CommandDescription => $"Attempts to open a job on the Grid Server via SoapUtility\nLayout:" +
                                            $"{MFDLabs.Grid.Bot.Properties.Settings.Default.Prefix}openjob jobID placeID?=1818 universeID?=1.";
        public string[] CommandAliases => new[] { "oj", "openjob" };
        public bool Internal => true;
        public bool IsEnabled { get; set; } = true;

        public async Task Invoke(string[] messageContentArray, SocketMessage message, string originalCommand)
        {
            if (!await message.RejectIfNotAdminAsync()) return;

            var jobId = messageContentArray.ElementAtOrDefault(0);
            if (!long.TryParse(messageContentArray.ElementAtOrDefault(1), out var placeId)) 
                placeId = 1818;
            if (!long.TryParse(messageContentArray.ElementAtOrDefault(2), out var universeId)) 
                universeId = 1;

            if (jobId == default)
            {
                await message.ReplyAsync($"Missing required parameter 'jobId', the layout is: " +
                                         $"{MFDLabs.Grid.Bot.Properties.Settings.Default.Prefix}{originalCommand} " +
                                         $"jobID placeID?=1818 universeID?=1`");
                return;
            }

            await GridServerCommandUtility.LaunchSimpleGameAsync(jobId, placeId, universeId);
            await message.ReplyAsync($"Successfully opened job '{jobId}' with placeID {placeId} and universeID {universeId}.");
        }
    }
}

File: Source/Lib/Commands/OpenJob.ts
import { Message } from 'discord.js';
import { GlobalConfig } from '../Utility/GlobalConfig';
import { AdminUtil } from '../Administration/AdminUtil';
import { ConversionUtil } from '../Utility/ConversionUtil';
import { RccCommandsUtil } from '../RCCService/RccCommandsUtil';
import { ScriptUtil } from '../RCCService/ScriptUtility';
import { StateSpecificCommandHandler } from '../CommandUtility/Types/StateSpecificCommandHandler';

class OpenJob implements StateSpecificCommandHandler {
	public CommandName = 'Open Job';
	public CommandDescription = 'Opens a job with the parameters of jobID, placeId and gameID';
	public Internal = true;
	public Command = ['oj', 'openjob', 'opengame'];
	public Callback = async (messageContent: string[], messageRoot: Message) => {
		if (!AdminUtil.RejectIfNotAdmin(messageRoot)) return;

		const jobId = messageContent[0];
		let placeId = ConversionUtil.ToInt64(messageContent[1]);
		let universeId = ConversionUtil.ToInt64(messageContent[2]);

		if (!jobId)
			return messageRoot.reply(
				`Missing required parameter 'jobId', the layout is: ${GlobalConfig.Prefix}openjob jobID placeID?=1818 universeID?=1`,
			);

		if (isNaN(placeId)) placeId = 1818;
		if (isNaN(universeId)) universeId = 1;

		return ScriptUtil.ParseLuaValuesAndRespond(await RccCommandsUtil.LaunchSimpleGame(jobId, placeId, universeId), messageRoot);
	};
}

export = new OpenJob();
